pkg/rules/echo/echo_server_setup.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2024 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//go:build ignore
+
+package rule
+
+import (
+	"strconv"
+
+	echo "github.com/labstack/echo/v4"
+)
+
+func Server() echo.MiddlewareFunc {
```

Comment:
Maybe it is better to rename it to `otelTraceMiddleware` and do not make it public